src/topology/continuous_function/t0_sierpinski.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2022 Ivan Sadofschi Costa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ivan Sadofschi Costa
+-/
+import topology.order
+import topology.continuous_function.basic
+
+/-!
+# Any T0 space embeds in a product of copies of the Sierpinski space
+
+We consider `Prop` with the Sierpinski topology. If `(X, t)` is a topological space, there is a
+continuous map `i : X → Π u ∈ t, Prop` defined as the product of the maps `iᵤ : X → Prop` defined by
+`iᵤ x = x ∈ u`. The map `i` is always inducing. Whenever `X` is T0, `i` is also injective and
+therefore an embedding.
+-/
+
+noncomputable theory
+
+variables {X : Type*} [t : topological_space X] [t0 : t0_space X]
+
+instance : topological_space (Π (u : {s : set X | t.is_open s}), Prop) :=
+@Pi.topological_space _ (λ u, Prop) (λ u, sierpinski_space)
+
+/--
+  The continuous map from `X` to the product of copies of the Sierpinski space, (one copy for each
+  open subset `u` of `X`). The `u` coordinate of  `i x` is given by `x ∈ u`.
+-/
+def i (X : Type*) [t : topological_space X] : continuous_map X
```

Comment:
This should be given a more descriptive name.

Reply:
Done! I'm not sure if `product_of_mem_opens` is a reasonable name. What do you think?

Reply:
I prefer prefixing functions by `to_`, but that's up to you. The name seems fine.